### ISSUE TITLE
[bug] Unable to download bactopia

### ISSUE THREAD

**Issue by baileybowcutt** [bactopia/bactopia#355]
**Description**
I am trying to download bactopia to my user home directory on my university's cluster computing. I have anaconda downloaded. I tried many different conditions, and I always seem to get it stuck loading on "Solving environment", for up to 4 hours when submitted as a job. It was able to install successfully on my local computer.

**Steps to Reproduce**
conda create -n bactopia -c conda-forge -c bioconda bactopia
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: | 

I tried talking to my university's hpcc help and they also couldn't figure out how to install it. We tried changing conda channel priority (strict and flexible and false did not work), doing module purge and module load conda/3, doing it on miniconda, making a separate conda environment first and loading it in there, doing it in a research/scratch space, and doing it on an interactive dev-node vs a SLURM job submission. Conda is 4.14.0. I also tried updating conda and it didn't help. It worked just fine when I did it locally. 

Thank you!

**Comment by rpetit3** [comment]
Hi @baileybowcutt,

I think best bet might be to try installing it with [Mamba](https://mamba.readthedocs.io/en/latest/user_guide/mamba.html), which should be must faster.

Mamba is ok to install in your base conda environment, so something like this:
```
conda deactivate # If in another env
conda install -c conda-forge -c bioconda mamba
```

Once mamba is installed:
```
mamba create -n bactopia -c conda-forge -c bioconda bactopia
conda activate bactopia
```

If this doesn't work, let me know! 

Once its working, and you want to start running on your cluster, I'm happy to work together to set up a profile for your cluster. Here's an example of one I'm using here: [ARCC](https://github.com/bactopia/bactopia/blob/dev/conf/profiles/arcc.config).

This lets me just add `-profile arcc` and Nextflow handles everything else.

Cheers, 
Robert

**Comment by baileybowcutt** [comment]
Thank you for the quick response! When download mamba I get:
```
conda install -c conda-forge -c bioconda mamba
Collecting package metadata (current_repodata.json): done
Solving environment: done

# All requested packages already installed.

Retrieving notices: ...working... done

Which looks great. 

```
When I move forward to run it I get the following error. Thoughts?
```
mamba create -n bactopia -c conda-forge -c bioconda bactopia
Traceback (most recent call last):
  File "/mnt/home/bowcuttb/anaconda3/bin/mamba", line 11, in <module>
    sys.exit(main())
  File "/mnt/home/bowcuttb/anaconda3/lib/python3.9/site-packages/mamba/mamba.py", line 848, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (/mnt/home/bowcuttb/anaconda3/lib/python3.9/site-packages/conda/common/compat.py)
```

**Comment by rpetit3** [comment]
What do you get with `mamba --version`?

**Comment by baileybowcutt** [comment]
I get: 
 ```
mamba --version
Traceback (most recent call last):
  File "/mnt/home/bowcuttb/anaconda3/bin/mamba", line 11, in <module>
    sys.exit(main())
  File "/mnt/home/bowcuttb/anaconda3/lib/python3.9/site-packages/mamba/mamba.py", line 848, in main
    from conda.common.compat import ensure_text_type, init_std_stream_encoding
ImportError: cannot import name 'init_std_stream_encoding' from 'conda.common.compat' (/mnt/home/bowcuttb/anaconda3/lib/python3.9/site-packages/conda/common/compat.py)
```

**Comment by rpetit3** [comment]
Oh my, that's not helpful! How about `conda env export | grep mamba` ?

**Comment by baileybowcutt** [comment]
I get the following:
`conda env export | grep mamba
  - mamba=0.7.6=py39heb6f48a_0
`

**Comment by rpetit3** [comment]
I think that might be our issue, Conda installed an older version of mamba. 

Lets try: `conda install -c conda-forge 'mamba>=0.24.*'` Hopefully conda will install a new version

**Comment by baileybowcutt** [comment]
Sorry about the delay. Trying that, but it just seems to be spinning. I'll give it more time, but I thought I'd share in case that's unusual and I should just kill it. 
```
conda install -c conda-forge 'mamba>=0.24.*'
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: - 
```

**Comment by baileybowcutt** [comment]
Update: It got stuck on the spinning wheel again. I tried again, and ended up getting the following:
<img width="1300" alt="IMG_1343" src="https://user-images.githubusercontent.com/72410151/191150246-4fa0dd94-b33e-4e71-aac9-122b7ef5468a.png">
I think it may have hit the 2 hour limit of our interactive dev-nodes, unless that is the full error. I resubmitted it as a job for 3.5 hours to see if it would produce the same result. 

I know downgrading conda is really not recommended, but do you think the current version of bactopia would run correctly with conda=4.13.0? This is the version I have on my local computer that was able to download bactopia easily yesterday. I could try to switch from 4.14.0 to 4.13.0 for the sake of getting the program installed.

**Comment by baileybowcutt** [comment]
Update: after 3.5 hours, the update mamba job timed out. It had the same code as above, with the addition of large sections examining for conflict:
```
Examining conflict for h5py cached-property anaconda:   0%|          | 0/399 [00:00<?, ?it/s]
g conflict for cached-property soupsieve atomicwrites h5py sphinxcontrib-htmlhelp zope singledispatch sqlite widgets
nbextension xlwt pysocks intervaltree urllib3 backports imagecodecs autopep8 backports.functools_lru_cache libcurl m
atplotlib-base numpy matplotlib-inline doc
```
Any other thoughts? I'll go into the cluster help hours again and let you know if they think of anything.

**Comment by rpetit3** [comment]
What does you're base environment look like on the cluster? `conda env export` 

I haven't tried things with anaconda, usually just miniconda.

**Comment by baileybowcutt** [comment]
Thank you for continuing to help with this. I have miniconda, and tried it there also but got the same spinning wheel. Here's my code:
<img width="858" alt="image" src="https://user-images.githubusercontent.com/72410151/191325594-e5f1237d-5ce7-40de-9d48-3a21aaa52bb7.png">

I just reached out to the cluster staff again and they said another person was having problems with conda. It appears this may be a problem on their end, but I'll keep you updated.

**Comment by rpetit3** [comment]
That miniconda env looks nice and clean, I would try installing mamba into it

**Comment by baileybowcutt** [comment]
Update!! I was able to use miniconda, update mamba, and use your mamba code to get things running!! YAY!! Thank you for all the help!

The cluster computing staff here said they had some issues with ariba on our system, and I noticed when I downloaded my datasets using
```
 bactopia datasets \
    --verbose \
    --species "Escherichia coli" \
    --include_genus \
    --limit 100 \
    --cpus 4
```
My final folders were the following:
<img width="319" alt="image" src="https://user-images.githubusercontent.com/72410151/191607901-c605081b-98c4-44ec-987a-ca621721dced.png">

I seem to be lacking ariba, but I'll look into that separately from this issue. Thank you so much for the patience and all the help with this problem!

**Comment by rpetit3** [comment]
Awesome! Does your HPC make use of singualrity?

As far as Ariba goes, I moved it to a Bactopia Tool a few versions ago.  It was starting to cause a few headaches and I didn't want that to affect the whole pipeline.

**Comment by baileybowcutt** [comment]
Sorry for the late response, but yes my HPC does make use of singularity! I had some problems again with getting things started and everything really ran a lot smoother when I used that -profile singularity command.

**Comment by rpetit3** [comment]
Hi @baileybowcutt 

I've released [v2.2.0](https://github.com/bactopia/bactopia/releases/tag/v2.2.0), going to go ahead and close this.

Cheers,
Robert
